homepage/homepage/app/(docs)/docs/[framework]/[[...slug]]/project-setup/providers/react-native-expo.mdx

```diff
@@ -0,0 +1,143 @@
+      AccountSchema={MyAppAccount}
+    >
+      {children}
+    </JazzProvider>
+  );
+}
+
+// Register the Account schema so `useAccount` returns our custom `MyAppAccount`
+declare module "jazz-expo" {
+  interface Register {
+    Account: MyAppAccount;
+  }
+}
+```
+</CodeGroup>
+
+## Provider Options
+
+- `kvStore`
+  - `ExpoSecureStoreAdapter` (default)
+- `AccountSchema`
+  - `Account` (default)
+- `CryptoProvider`
+  - `PureJSCrypto` (default) - Pure JavaScript crypto provider
+  - `RNQuickCrypto` - C++ accelerated crypto provider
+
+## Authentication in the Provider
+
+`<JazzProvider />` works with various authentication methods, with Passkey being the easiest way to get started. Passkeys are secure, simple for users, and work across devices.
+
+For detailed implementation, refer to our [Authentication Overview](/docs/authentication/overview) guide.
+
+<CodeGroup>
+```tsx
+// Example with Passkey authentication
+import { JazzProvider, usePasskeyAuth } from "jazz-expo";
+
+function AuthenticatedProvider({ children }) {
+  const auth = usePasskeyAuth();
+
+  // Show passkey registration UI if needed
+  if (auth.state === "needsRegistration") {
+    return <PasskeyRegistrationScreen auth={auth} />;
+  }
+  
+  // Show passkey sign-in UI if needed
+  if (auth.state === "needsSignIn") {
+    return <PasskeySignInScreen auth={auth} />;
+  }
+  
+  return (
+    <JazzProvider
+      sync={{ 
+        peer: "wss://cloud.jazz.tools/?key=your-api-key",
+        // Only sync when authenticated
+        when: "authenticated" 
+      }}
+      AccountSchema={MyAppAccount}
+      auth={auth}
+    >
+      {children}
+    </JazzProvider>
+  );
+}
+```
+</CodeGroup>
+
+For a complete example, see the [Expo Chat Demo with Clerk](https://github.com/garden-co/jazz/tree/main/examples/chat-rn-expo-clerk).
+
+## Local Persistence
+
+Jazz for Expo includes built-in local persistence using SQLite. Following Expo's best practices, the Expo implementation uses:
+
+- **Database Storage**: `expo-sqlite` - Expo's official SQLite module
+- **Key-Value Storage**: `expo-secure-store` - Expo's secure storage system
+
+Local persistence is enabled by default with no additional configuration required. Your data will automatically persist across app restarts.
+
+### Advanced Configuration
+
+You can customize the storage options in the `JazzProvider`:
+
+<CodeGroup>
+```tsx
+// App.tsx
+import {
+  JazzProvider,
+  ExpoSQLiteAdapter,
+  ExpoSecureStoreAdapter,
+} from "jazz-expo";
+
+// Default configuration - no parameters needed
+<JazzProvider
+  sync={{ peer: "wss://cloud.jazz.tools/?key=you@example.com" }}
+  AccountSchema={MyAppAccount}
+>
+  {children}
+</JazzProvider>
+
+// Or with explicit configuration
+<JazzProvider
+  storage="sqlite" // Use default ExpoSQLiteAdapter
+  // Or with custom configuration
+  // storage={{
+  //   adapter: new ExpoSQLiteAdapter({ /* custom options */ }),
+  //   kvStore: new ExpoSecureStoreAdapter({ /* custom options */ })
+  // }}
```

Comment:
These storage options aren't valid.

`storage` takes `ExpoSQLiteAdapter` directly (which is the default) and the "sqlite" option is invalid.

`kvStore` is a separate prop, but I wonder if we could come up with a better name.

How about `credentialsStore`?

Reply:
I understand now. Yes, that makes sense, I've created a ticket for it #1799